### PR TITLE
Add support for powervs image list command

### PR DESCRIPTION
**Pull request by KeerthanaAP**
**What this PR does / why we need it**:

In this PR, a CLI command to list image  is added

```
 % ./capibmadm powervs image list --service-instance-id 5e1d357c-7603-42a9-b4c1-fe3772bdfa0c --zone lon04
Listing PowerVS images service-instance-id="5e1d357c-7603-42a9-b4c1-fe3772bdfa0c"
ID                                     NAME               STATE    DESCRIPTION   STORAGE POOL    STORAGE TYPE   CREATION DATE              LAST UPDATE DATE           ARCH    CONTAINER FORMAT   DISK FORMAT   ENDIANNESS      HYPERVISOR TYPE    OS     IMAGE TYPE
06ff0390-22e1-41fe-9324-3d637bcdb6f1   CentOS-Stream-8    active                 Tier3-Flash-1   tier3          2023-02-16T15:00:00.000Z   2023-02-16T15:01:31.000Z   ppc64   bare               raw           little-endian   phyp               rhel   stock
47f2f988-d2ea-4cb6-922a-c895bf45f5f4   Linux-CentOS-8-3   active                 Tier3-Flash-1   tier3          2023-02-16T14:57:05.000Z   2023-02-16T14:58:54.000Z   ppc64   bare               raw           little-endian   phyp               rhel   stock
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/capibmadm/cmd/powervs/image/type.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+package image
+
+import (
+	"github.com/go-openapi/strfmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ISpec defines a Image.
+type ISpec struct {
```
Comment (Amulyam24) [inline cmd/capibmadm/cmd/powervs/image/type.go:26]: @KeerthanaAP, how about a more meaningful name such as `ImageSpec` and `ImageList`?

File: cmd/capibmadm/cmd/powervs/image/list.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	if len(images.Images) == 0 {
+		fmt.Println("No images found")
+		return nil
+	}
+
+	imageList := IList{
+		Items: []ISpec{},
+	}
+
+	for _, image := range images.Images {
```
Comment (Amulyam24) [inline cmd/capibmadm/cmd/powervs/image/list.go:87]: You might need to take care of paging here for listing all available images. For reference you can check this -
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/cmd/capibmadm/cmd/vpc/image/list.go#L79-L104

Feel free to reach out to me for any questions.

Comment (Amulyam24) [reply cmd/capibmadm/cmd/powervs/image/list.go:87]: Sorry, irrelevant for PowerVS, can be ignored.

File: cmd/capibmadm/cmd/powervs/image/type.go (hunk 1 above)
Comment (KeerthanaAP) [reply cmd/capibmadm/cmd/powervs/image/type.go]: Done. As lint is throwing the below error  when using `ImageSpec` and `ImageList` names, changed names to `ImgSpec` and `ImgList`.


Error when using `ImageSpec` and `ImageList` names:

`exported: type name will be used as image.ImageList by other packages, and that stutters; consider calling this List (revive)`

File: cmd/capibmadm/cmd/powervs/image/list.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+			imageToAppend.HypervisorType = image.Specifications.HypervisorType
+			imageToAppend.ImageType = image.Specifications.ImageType
+			imageToAppend.OperatingSystem = image.Specifications.OperatingSystem
+		}
+
+		imageList.Items = append(imageList.Items, imageToAppend)
+	}
+
+	printerObj, err := printer.New(options.GlobalOptions.Output, os.Stdout)
+
```
Comment (Amulyam24) [inline cmd/capibmadm/cmd/powervs/image/list.go:112]: extra line can be omitted here.

**Review by Amulyam24 — COMMENTED** [review]
Apart from minor comment, LGTM!

**Pull request merged**